### ISSUE TITLE
Add Prometheus exemplar support to Heatmaps

### ISSUE THREAD

**Issue by bwplotka** [grafana/grafana#32917]
**What would you like to be added**:

Currently, Prometheus exemplars are supported on TimeSeries panels. It would be nice to leverage them on heatmaps too! 🤗 

**Why is this needed**:

To enhance the experience of working with exemplars on panels with different forms.

**Comment by davkal** [comment]
The heatmap panel is being rewritten soon. Once that is done, we can get on this.

**Comment by srclosson** [comment]
@davkal The heatmap panel is in alpha now. But I have some thoughts about this. Will ping you.

**Comment by metalmatze** [comment]
Technically I can point out that the OpenMetrics standard (which is always used with exemplars) will always have a bucket for an examplar: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#exemplars-1
If a bucket exists, it can optionally have an exemplar. 

So the black dots on the heatmap that don't lay over a particular bucket won't be possible in practice.

**Comment by srclosson** [comment]
Good to know. @metalmatze

**Comment by leeoniya** [comment]
we merged support for this into `main` a while ago. still some rendering improvements to do, but we'll tackle that in a separate issue.

![image](https://user-images.githubusercontent.com/43234/172007139-2d577aa2-2bb8-413e-83d9-c0be3b5caf71.png)
